From 4bdee849da4bd6118266548f28a75e1222d4b996 Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Tue, 16 Sep 2014 15:04:41 -0400
Subject: [net] vxlan: fix potential NULL dereference in arp_reduce()

Message-id: <d676878c09bbc0234062d677c0ae260ea7555522.1410441647.git.dborkman@redhat.com>
Patchwork-id: 95082
O-Subject: [RHEL7.1 PATCH net 4/7] vxlan: fix potential NULL dereference in arp_reduce()
Bugzilla: 1130643
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1130643
Upstream Status: 7346135dcd3f9b57f30a5512094848c678d7143e
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7947483
Tested: verified by IBM and myself

Upstream Commit:
commit 7346135dcd3f9b57f30a5512094848c678d7143e
Author: David Stevens <dlstevens@us.ibm.com>
Date:   Tue Mar 18 12:32:29 2014 -0400

    vxlan: fix potential NULL dereference in arp_reduce()

    This patch fixes a NULL pointer dereference in the event of an
    skb allocation failure in arp_reduce().

    Signed-Off-By: David L Stevens <dlstevens@us.ibm.com>
    Acked-by: Cong Wang <cwang@twopensource.com>

    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/vxlan.c b/drivers/net/vxlan.c
index 7dd99cc..0c8363b 100644
--- a/drivers/net/vxlan.c
+++ b/drivers/net/vxlan.c
@@ -1293,6 +1293,9 @@ static int arp_reduce(struct net_device *dev, struct sk_buff *skb)
 
   neigh_release(n);
 
+  if (reply == NULL)
+   goto out;
+
   skb_reset_mac_header(reply);
   __skb_pull(reply, skb_network_offset(reply));
   reply->ip_summed = CHECKSUM_UNNECESSARY;
-- 
1.7.1